/*
Copyright 2026 Man Group Operations Limited

Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.

As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
*/
syntax = "proto3";

package arcticc.pb2.s3_storage_pb2;

message Config {
    string bucket_name = 1;
    string credential_name = 2;
    string credential_key = 3;
    string endpoint = 4;
    uint32 max_connections = 5;
    uint32 connect_timeout = 6;
    uint32 request_timeout = 7;
    bool ssl = 8;
    string prefix = 9;
    bool https = 10;
    string region = 11;
    bool use_virtual_addressing = 12;
    bool use_mock_storage_for_testing = 13;
    string ca_cert_path = 14;
    string ca_cert_dir = 15;
    // Whether to leave the prefix unchanged, or to translate the periods in it.
    bool use_raw_prefix = 16;
}
